### PR TITLE
refactor(date-block): upgrade component - INNO-608

### PR DESCRIPTION
**Pull request by emeryro**
# PR description

Upgrade component.
Mainly css (no js).

# QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [x] `package.json` is up-to-date and `@ec-europa/ecl-base` is part of the dependencies
- [x] I have given the fractal status “ready” to my component
- [x] I have declared `@define mycomponent` in the SCSS file
- [x] I have specified `margin: 0;` on the CSS component
- [x] I have provided tests
- [x] I follow the naming guidelines
- [x] the component supports composition
- [x] the template is fully functional
- [x] I have filled the README.md file (at least a few lines)
- [x] I have checked the dependencies
- [x] there are no hardcoded strings (all content come from the context)

### CODE REVIEW

File: framework/components/ecl-date-blocks/ecl-date-blocks.scss [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    top: 0;
+  }
+}
+
+.ecl-date-block__body {
+  align-items: center;
+  display: inline-flex;
+  flex-direction: column;
+  justify-content: center;
+  min-height: map-get($ecl-spacing, 'xxxl');
```
Comment (yhuard) [inline framework/components/ecl-date-blocks/ecl-date-blocks.scss]: Height and width are about sizing, not spacing. You can use some kind of "magic" numbers if necessary

File: framework/base/variables/_spacing.scss [hunk 2]
```diff
@@ -9,5 +9,6 @@ $ecl-spacing: (
   m: 1.66rem,
   l: 2.22rem,
   xl: 3.33rem,
-  xxl: 4.44rem
+  xxl: 4.44rem,
+  xxxl: 5.55rem
```
Comment (yhuard) [inline framework/base/variables/_spacing.scss]: Please remove it if it's not used

File: framework/components/ecl-date-blocks/ecl-date-blocks.config.js [hunk 3]
```diff
@@ -1,67 +1,57 @@
...
-        slot3_modifier: 'month',
-        slot3_data: 'Oct',
-        slot4_modifier: 'day-text',
-        slot4_data: 'Mon-Wed',
-        slot5_modifier: 'day',
-        slot5_data: '04-06',
-        slot6_modifier: 'month',
-        slot6_data: 'Apr',
+        variant: '',
+        weekDay: 'Tue',
```
Comment (yhuard) [inline framework/components/ecl-date-blocks/ecl-date-blocks.config.js]: Does not respect the twig naming convention

File: framework/components/ecl-date-blocks/ecl-date-blocks.scss [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  flex-direction: column;
+  justify-content: center;
+  min-height: 5.55rem;
+  min-width: 5.55rem;
+}
+
+.ecl-date-block__week-day {
+  font-size: map-get($ecl-font-size, 'xs');
+  line-height: 1.2;
+  z-index: map-get($ecl-z-index, 'highlight');
```
Comment (yhuard) [inline framework/components/ecl-date-blocks/ecl-date-blocks.scss]: Out of curiosity, why do you have to use `z-index` here?

File: framework/components/ecl-date-blocks/ecl-date-blocks.twig [hunk 5]
```diff
@@ -1,13 +1,56 @@
...
+{#
+  - variant: string; display variant (default: '')
+  - weekDay: string; day(s) of the week of the event (default: '')
+  - day: string; day number(s) of the event (default: '')
+  - month: string; month(s) of the event (default: '')
+  - year: string; year(s) of the event (default: '')
+#}
+
+{% set variant = variant|default('') %}
+{% set weekDay = weekDay|default('') %}
```
Comment (yhuard) [inline framework/components/ecl-date-blocks/ecl-date-blocks.twig]: Please follow the twig convention for naming variables :)

File: framework/components/ecl-date-blocks/ecl-date-blocks.scss (hunk 4 above)
Comment (emeryro) [inline framework/components/ecl-date-blocks/ecl-date-blocks.scss]: Well it was needed at some point, but I refactored my code, so I guess this is useless now.
Will remove it.

**Comment by yhuard** [comment]
Will re-check

**Pull request merged**